src/gatsby/src/components/park/parkHeader.js

```diff
@@ -130,16 +130,16 @@ export default function ParkHeader({
               {isParkOpen !== false &&
                 <p>
                   The {parkType} {parkOperation?.hasParkGate !== false && "gate"} is open {parkDates}
-                  {(parkOperation?.gateOpenTime && parkOperation?.gateCloseTime) ? (
-                    parkOperation.gateOpenTime === "00:00:00" && parkOperation.gateCloseTime === "23:59:00" ? (
+                  {(parkOperation?.gateOpenTime || parkOperation?.gateCloseTime || parkOperation?.gateOpensAtDawn || parkOperation?.gateClosesAtDusk) ? (
+                    parkOperation?.gateOpenTime === "00:00:00" && parkOperation?.gateCloseTime === "23:59:00" ? (
                       <>, 24 hours a day.</>
                     ) : (
                       <>
                         , from <span className="no-wrap">
-                          {parkOperation?.gateOpensAtDawn ? "dawn" : formattedTime(parkOperation.gateOpenTime)}
+                          {parkOperation?.gateOpensAtDawn ? "dawn" : formattedTime(parkOperation?.gateOpenTime)}
```

Comment:
If `parkOperation.gateOpenTime` will ever be null/undefined (I see the added `?`) then it won't render anything and it might say "..is open  to  , daily."

Is there any more conditional logic we should sort out in the code before we print it in the template?

Reply:
Looking at the big block of conditions on lines 133-134, maybe we should move all of that up into the code and set a variable or two for the gate times to put into the template?

Reply:
My bad 😅 The logic is that if either `gateOpenTime` or `gateOpensAtDawn` is available, then either `gateCloseTime` or `gateClosesAtDusk` must also be available. And if one of them is missing, display a period ".".